ci-operator/step-registry/cucushift/upgrade/toimage/cucushift-upgrade-toimage-commands.sh

```diff
@@ -41,6 +41,78 @@ EOF
     fi
 }
 
+# Update RHEL repo before upgrade
+function rhel_repo(){
+    echo "Updating RHEL node repo"
+    # Ensure our UID, which is randomly generated, is in /etc/passwd. This is required
+    # to be able to SSH.
+    if ! whoami &> /dev/null; then
+        if [[ -w /etc/passwd ]]; then
+            echo "${USER_NAME:-default}:x:$(id -u):0:${USER_NAME:-default} user:${HOME}:/sbin/nologin" >> /etc/passwd
+        else
+            echo "/etc/passwd is not writeable, and user matching this uid is not found."
+            exit 1
+        fi
+    fi
+    SOURCE_REPO_VERSION=$(echo "${SOURCE_VERSION}" | cut -d'.' -f1,2)
+    TARGET_REPO_VERSION=$(echo "${TARGET_VERSION}" | cut -d'.' -f1,2)
+    export SOURCE_REPO_VERSION
+    export TARGET_REPO_VERSION
+
+    cat > repo.yaml <<-'EOF'
+---
+- name: Update repo Playbook
+  hosts: new_workers
+  any_errors_fatal: true
+  gather_facts: false
+  vars:
+    source_repo_version: "{{ lookup('env', 'SOURCE_REPO_VERSION') }}"
+    target_repo_version: "{{ lookup('env', 'TARGET_REPO_VERSION') }}"
+  tasks:
+  - name: Wait for host connection to ensure SSH has started
+    wait_for_connection:
+      timeout: 600
+  - name: Find all of the files inside /etc/yum.repos.d/
+    find:
+      paths: "/etc/yum.repos.d/"
+      patterns: "*.repo"
+    register: repos
+  - name: Replace source release version with target release version in the files
+    replace:
+      path: "{{ item.path }}"
+      regexp: "{{ source_repo_version }}"
+      replace: "{{ target_repo_version }}"
+    with_items: "{{ repos.files }}"
```

Comment:
Better to run a `yum clean all` to clean the previous repo caches after updating the yum repos.

Reply:
Good point! I'm thinking that without yum cache cleaning might be a valid scenario. I mean customers might not clean the old caches after changing the repos and our rhel upgrade playbook should be able to handle it. WDYT?

Reply:
The RHEL upgrade playbook doesn't clean up the yum cache before upgrading the packages, the first time it calls yum command should be here:
https://github.com/openshift/openshift-ansible/blob/master/roles/openshift_node/tasks/install.yml#L65-L68

I guess it mainly depends on how user is updating their yum repo when they upgraded the RHEL workers. 
If they only updated the existing repo baseurl to a new one, and without new yum cache made, the above step won't get the expected new rpm package. 
If they disabled the previous ocp repo and created a new one with a different channel/repo name, just like what we recommended in the official doc
```
subscription-manager repos --disable=rhel-7-server-ose-4.9-rpms --enable=rhel-7-server-ose-4.10-rpms 
```
it will make new cache automatically when calling yum command. 
So to make our automation more stable, I'll prefer adding a yum cache cleaning up step after replacing the 4.x keyword in the repo file.

Reply:
Thanks for the explanation! It's fine with me to clean up the cache.